.gitignore

```diff
@@ -10,3 +10,8 @@ src/files/*
 *.phpproj
 *.sln
 *.phpproj.user
+
+# for docker stuff
+/db/
+data/conf.php
```

Comment:
I guess the best solution for this would be to create an instruction for people to copy the conf.template.php from the inc folder to the data folder as conf.php

Reply:
So, there are a few reasons I created the empty conf.php file.

1) whenever you do a `docker-compose up` if you don't have a file there ( by default ) it will create a directory in the location you specify for a bind mount.
2) Having it be empty also allows the normal install process of hashtopolis to install since it can't access the DB ( at least from my brief interactions with the application that is what I derived ).
3) Then after the user puts in their information the empty file, that is bind mounted, gets overwritten with their config file.

Which leads to why I am ignoring it, so that way the empty file stay being tracked but no ones sensitive credentials that overwrite it are tracked.

>I guess the best solution for this would be to create an instruction for people to copy the conf.template.php from the inc folder to the data folder as conf.php

Wouldn't this require people also fill in their config information then? Compared to just letting them go through the web app install which will do the same thing, or am I missing something?

Reply:
I am not sure what the right way is for this, but now having it empty in there will force git to keep track of changes to it (even if you leave it in gitignore). Resulting in the posibility of credentials being push, especially for me while developing :D.

> Wouldn't this require people also fill in their config information then? Compared to just letting them go through the web app install which will do the same thing, or am I missing something?
No, the template is a placeholder and will still trigger the installation. But either copying the template or touching an empty file will result in the same.

Maybe you have a different solution for this problem? How are other project dealing with single configsfile that you want to make persistent?

Reply:
ah... I hadn't run `git status` yet on the place I deployed the docker stuff to :sweat_smile: 

IIRC they normally don't just bind mount a single file, but just bind mount a whole directory and put the config(s) in there. I don't think hashtopolis can do that though, since it has the config around a bunch of other files.

What we could do is have instructions somewhere that tells people once they `git clone` they can run [this](https://stackoverflow.com/a/23673910) to ensure their changes don't get tracked: `git update-index --assume-unchanged data/conf.php`

Reply:
closing this based on this change: https://github.com/hashtopolis/server/pull/719#discussion_r707993696
and removed the file from the repo: https://github.com/elreydetoda/hashtopolis-server/commit/c01baf9ec5d5a4bbb98c724deb697d4ff894c39c